internal/controller/vrg_volrep.go

```diff
@@ -137,20 +148,33 @@ func (v *VRGInstance) reconcileVolRepsAsSecondary() bool {
 			continue
 		}
 
+		vrMissing, requeueResult := v.reconcileMissingVR(pvc, log)
+		if vrMissing || requeueResult {
+			requeue = true
+
+			continue
+		}
+
 		// If VR is not ready as Secondary, we can ignore it here, either a future VR change or the requeue would
 		// reconcile it to the desired state.
-		requeueResult, _, skip = v.reconcileVRAsSecondary(pvc, log)
+		requeueResult, ready, skip := v.reconcileVRAsSecondary(pvc, log)
 		if requeueResult {
 			requeue = true
 
 			continue
 		}
 
-		if skip {
+		if skip || !ready {
 			continue
 		}
 
-		log.Info("Successfully processed VolumeReplication for PersistentVolumeClaim")
+		if v.undoPVCFinalizersAndPVRetention(pvc, log) {
+			requeue = true
+
+			continue
+		}
+
+		v.pvcStatusDeleteIfPresent(pvc.Namespace, pvc.Name, log)
```

Comment:
This is a good place to delete the `ProtectedPVC` for this PVC. However, reaching this point indicates that line 171 was successful and the PV/PVC were updated, but there is still a possibility that the VRG update might fail. See the comment above for more context.